src/compiler/moduleSpecifiers.ts

```diff
@@ -832,17 +847,64 @@ const enum MatchingMode {
     Pattern,
 }
 
-function tryGetModuleNameFromExports(options: CompilerOptions, targetFilePath: string, packageDirectory: string, packageName: string, exports: unknown, conditions: string[], mode = MatchingMode.Exact): { moduleFileToTry: string; } | undefined {
+function getOutputPath(targetFilePath: string, options: CompilerOptions, commonSourceDirectory: string) {

```

Comment:
Helpers for this are present too: 
`getOutputJSFileName`

Reply:
see https://github.com/microsoft/TypeScript/pull/55015#discussion_r1376855262

Reply:
I think that can be modified with signature where getCommonSourceDirectory is passed and doesnt need fileNames but just options ...  we dont want two methods for sure to do same thing and result in mismatch/issues. (we did have them before merging all these things together)